From 7ec7d1eee022e629397bf64938b2948b9537bb46 Mon Sep 17 00:00:00 2001
From: Michael Sweet <michael.r.sweet@gmail.com>
Date: Mon, 28 Aug 2017 10:04:29 -0400
Subject: Fix an Avahi-related crash bug in the scheduler (Issue #5085, Issue
 #5086)

Add NULL pointer check to avoid Avahi assertion in production code.

Bug: https://github.com/apple/cups/issues/5085
Bug: https://github.com/apple/cups/issues/5086
Bug-Ubuntu: https://launchpad.net/bugs/1442169
Bug-Ubuntu: https://launchpad.net/bugs/1495276
Patch-Name: scheduler-fix-avahi-crashes.patch
---
 scheduler/dirsvc.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/scheduler/dirsvc.c b/scheduler/dirsvc.c
index 2612d4d2f..ad48b15aa 100644
--- a/scheduler/dirsvc.c
+++ b/scheduler/dirsvc.c
@@ -676,13 +676,16 @@ dnssdDeregisterInstance(
   DNSServiceRefDeallocate(*srv);
 
 #  else /* HAVE_AVAHI */
-  if (!from_callback)
-    avahi_threaded_poll_lock(DNSSDMaster);
+  if (*srv)
+  {
+    if (!from_callback)
+      avahi_threaded_poll_lock(DNSSDMaster);
 
-  avahi_entry_group_free(*srv);
+    avahi_entry_group_free(*srv);
 
-  if (!from_callback)
-    avahi_threaded_poll_unlock(DNSSDMaster);
+    if (!from_callback)
+      avahi_threaded_poll_unlock(DNSSDMaster);
+  }
 #  endif /* HAVE_DNSSD */
 
   *srv = NULL;
